### PR TITLE
JAVA-29296 Upgrade spring-security-oidc

### PR DESCRIPTION
**Pull request by timis1**

### CODE REVIEW

File: spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/jwtauthorities/config/SecurityConfig.java [hunk 1]
```diff
@@ -64,12 +64,9 @@ public Converter<Jwt,AbstractAuthenticationToken> customJwtAuthenticationConvert
     @Bean
     SecurityFilterChain customJwtSecurityChain(HttpSecurity http) throws Exception {
         // @formatter:off
-        return http.oauth2ResourceServer(oauth2 -> {
-            oauth2.jwt()
-              .jwtAuthenticationConverter(customJwtAuthenticationConverter(accountService));
-        })
+        return http.oauth2ResourceServer(oauth2 -> oauth2.jwt(jwtConfigurer -> jwtConfigurer
+                                .jwtAuthenticationConverter(customJwtAuthenticationConverter(accountService))))
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/jwtauthorities/config/SecurityConfig.java:68]: Can you convert this to three lines please and fix the indentations.

File: spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java [hunk 2]
```diff
@@ -19,19 +20,18 @@ public class OAuth2SessionManagementSecurityConfig {
 
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.authorizeRequests(authorizeRequests -> authorizeRequests.mvcMatchers("/home")
-            .permitAll()
-            .anyRequest()
-            .authenticated())
-            .oauth2Login(oauthLogin -> oauthLogin.permitAll())
+        http.authorizeHttpRequests(authorizeRequests -> authorizeRequests
+                        .requestMatchers("/home").permitAll()
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java:24]: Please fix the indentation.

File: spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java [hunk 3]
```diff
@@ -19,19 +20,18 @@ public class OAuth2SessionManagementSecurityConfig {
...
+            .oauth2Login(AbstractAuthenticationFilterConfigurer::permitAll)
             .logout(logout -> logout.logoutSuccessHandler(oidcLogoutSuccessHandler()));
         return http.build();
     }
 
     private LogoutSuccessHandler oidcLogoutSuccessHandler() {
         OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler = new OidcClientInitiatedLogoutSuccessHandler(this.clientRegistrationRepository);
 
-        oidcLogoutSuccessHandler.setPostLogoutRedirectUri(URI.create("http://localhost:8081/home"));
+        oidcLogoutSuccessHandler.setPostLogoutRedirectUri(URI.create("http://localhost:8081/home").toString());
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java:34]: I guess can be shortened to oidcLogoutSuccessHandler.setPostLogoutRedirectUri("http://localhost:8081/home");

**Review by alvin577austria — CHANGES_REQUESTED** [review]

File: spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java (hunk 3 above)
Comment (timis1) [reply spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/sessionmanagement/config/OAuth2SessionManagementSecurityConfig.java:34]: Fixed

File: spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/jwtauthorities/config/SecurityConfig.java (hunk 1 above)
Comment (timis1) [reply spring-security-modules/spring-security-oidc/src/main/java/com/baeldung/openid/oidc/jwtauthorities/config/SecurityConfig.java:68]: Fixed

**Review by alvin577austria — APPROVED** [review]

**Comment by alvin577austria** [comment]
Will merge when the current master branch build is green.

**Pull request merged**